src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -119,6 +121,7 @@ public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSp
     if (template != null) {
       properties.put(AribaPluginConfig.TEMPLATE_NAME, template);
     }
+    properties.put(Constants.Reference.REFERENCE_NAME, ReferenceNames.cleanseReferenceName(template));
```

Comment:
we should not pass arg null to ReferenceNames.cleanseReferenceName()

Reply:
Reworked on it.